### PR TITLE
Added integration test

### PR DESCRIPTION
**Pull request by ravit-db**
- Created the setup for Reconciliation Integration Test
-  Added integration tests for Databricks to Databricks data reconciliation

### CODE REVIEW

File: tests/integration/test_execute_databricks.py [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+    StatusOutput,
+    Filters,
+    Transformation,
+    Thresholds,
+)
+from tests.integration.test_utils import get_reports
+
+
+@pytest.fixture
+def setup_databricks_src(setup_teardown, spark, test_config):
```
Comment (sundarshankar89) [inline tests/integration/test_execute_databricks.py:33]: setup can me moved into setup.py or conf.py

File: tests/integration/test_execute_databricks.py [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+    src_data.write.format("delta").mode("overwrite").saveAsTable(
+        f"{test_config.db_mock_catalog}." f"{test_config.db_mock_schema}.{test_config.db_mock_src}"
+    )
+    tgt_data.write.format("delta").mode("overwrite").saveAsTable(
+        f"{test_config.db_mock_catalog}." f"{test_config.db_mock_schema}.{test_config.db_mock_tgt}"
+    )
+
+
+def test_execute_report_type_is_data_with_all_match(setup_databricks_src, spark, ws, test_config, reconcile_config):
```
Comment (sundarshankar89) [inline tests/integration/test_execute_databricks.py:240]: where are all the scenarios we are testing?

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

File: tests/integration/test_execute_databricks.py (hunk 2 above)
Comment (ravit-db) [reply tests/integration/test_execute_databricks.py:240]: It has all four report types and has tested all the configurations. Let me know if anything needs to be added or if something should be covered.

Comment (sundarshankar89) [reply tests/integration/test_execute_databricks.py:240]: we should be testing other configuration like scenarios, complex types, thresholds, transformation, filters etc.

File: tests/integration/test_execute_databricks.py (hunk 1 above)
Comment (ravit-db) [reply tests/integration/test_execute_databricks.py]: fixed.

File: .github/workflows/acceptance.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  integration:
+    if: github.event_name == 'pull_request' && github.event.pull_request.draft == false
+    environment: account-admin
```
Comment (nfx) [inline .github/workflows/acceptance.yml:24]:
```suggestion
    environment: tool
```

Comment (nfx) [reply .github/workflows/acceptance.yml:24]: this is a correct scope for remorph, not account-admin.

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+    return DatabricksSession.builder.sdkConfig(ws.config).getOrCreate()
+
+
+def _get_metrics_config() -> TestMetricsConfig:
+    return TestMetricsConfig(catalog="integration_test", schema="reconcile", volume="test_volume")
+
+
+def _get_test_config() -> TestConfig:
+    return TestConfig(
+        db_table_catalog="samples",
```
Comment (nfx) [inline tests/integration/conftest.py:63]: keep in mind that all catalogs are wiped out by our test infrastructure, so you have to pre-create randomly-named catalogs for every tests and tear them down. actually... let's create a test catalog just for you

**Review by nfx — CHANGES_REQUESTED** [review]

File: .github/workflows/acceptance.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          cache: 'pip'
+          cache-dependency-path: '**/pyproject.toml'
+          python-version: '3.10'
+
+      - name: Install hatch
+        run: pip install hatch==1.9.4
+
+      - name: Run integration tests
+        uses: databrickslabs/sandbox/acceptance@acceptance/v0.2.2
```
Comment (nfx) [inline .github/workflows/acceptance.yml:43]: you also need `acceptance_path` in the `.codegen.json` https://github.com/databrickslabs/ucx/blob/main/.codegen.json#L9

Comment (ravit-db) [reply .github/workflows/acceptance.yml:43]: added

File: tests/integration/conftest.py (hunk 4 above)
Comment (ravit-db) [reply tests/integration/conftest.py:63]: tear down process is already in place to cleanup the catalog. Let me know if I missed out anything or any alternatives on that.

**Comment by sundarshankar89** [comment]
too divergent will refactor and recreate a new one.

**Pull request closed without merging**